network.py

```diff
@@ -181,7 +188,17 @@ def close(self):
             del self._token
         if hasattr(self, u'_master_token'):
             del self._master_token
-        self._session = None
+        sessions = list(self._active_sessions)
+        if sessions:
+            self.logger.warn("Closing %d active sessions" % len(sessions))
```

Comment:
I only avoid that syntax because it only works for logging functions and I sometimes switch back and forth with other printers like print().  But I can use printf style formatting from now on.